core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -122,6 +119,18 @@ class ShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
    */
   private[this] var _shuffleMergedFinalized: Boolean = false
 
+  /**
+   * shuffleSequenceId is used to give temporal ordering to the executions of a ShuffleDependency.
+   * This is required in order to handle indeterministic stage retries for push-based shuffle.
+   */
+  private[this] var _shuffleSequenceId: Int = -1
```

Comment:
How about renaming it as `_shuffleMergeSequenceId ` or just `_ shuffleMergeId `? Because I'm thinking that `_shuffleSequenceId` won't increase for a determinate stage. But `_shuffleSequenceId` just sounds like it'd increase whenever the ShuffleDependency is re-executed, even if it's a determinate stage.

Reply:
Yes, you're correct. Currently for determinate stage we don't increment the `shuffleSequenceId`. Currently we disable merge in the case when shuffle is finalized. But I can think of one case where we can do in the future which is in a retry attempt of a stage for the same shuffle ID we can generate new set of files and have both the older and newer attempt blocks to be read on the reducer side effectively increasing the overall merged blocks read. Thoughts?

Reply:
It sounds like the idea is opposite to the current solution for the indeterminate stage (where an older fetch/push always fails), so I imagine it would introduce many inconsistent changes for the determinate stage, which could be troublesome.

Bseides, if we want to read data from older and newer attempts together, IIUC, reset ShuffleDependency._shuffleMergedFinalized to false would allow data from newer attempts to be merged into the same merged files with older attempts, so we can also read them together. And this sounds like more easier to do.

Reply:
I see, makes sense. It kind of complicates things.